Add Config to Skip Cpp Codegen, Enable in FBCode

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #97198
* __->__ #97204



cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

Differential Revision: [D44353662](https://our.internmc.facebook.com/intern/diff/D44353662)

eellison committed:
Add Config to Skip Cpp Codegen, Enable in FBCode

[ghstack-poisoned]

bertmaher reviewed (approved):
Nice!! Thank you!

mikekgfb reviewed (approved):
So cool!   Thank you!

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44353662).

eellison commented:
@eellison has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44353662).

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44353662](https://our.internmc.facebook.com/intern/diff/D44353662)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44353662).

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44353662](https://our.internmc.facebook.com/intern/diff/D44353662)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44353662).

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44353662](https://our.internmc.facebook.com/intern/diff/D44353662)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44353662).

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44353662](https://our.internmc.facebook.com/intern/diff/D44353662)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44353662).

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44353662](https://our.internmc.facebook.com/intern/diff/D44353662)

[ghstack-poisoned]

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44353662](https://our.internmc.facebook.com/intern/diff/D44353662)

[ghstack-poisoned]

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44353662](https://our.internmc.facebook.com/intern/diff/D44353662)

[ghstack-poisoned]

eellison committed:
Update on "Add Config to Skip Cpp Codegen, Enable in FBCode"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44353662](https://our.internmc.facebook.com/intern/diff/D44353662)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44353662).

titaiwangms commented on torch/fx/passes/fake_tensor_prop.py:
```diff
@@ -29,10 +31,29 @@ def __init__(self, module: torch.fx.GraphModule, mode: Optional[FakeTensorMode]
+            elif isinstance(obj, py_sym_types):
+                return obj
+            else:
+                return None
```
I am experimenting that using symbolic mode of fx.graph with FakeTensorProp to get shapes, and further use it to validate torch/onnx op level results inside a dynamic model. Before this PR, the obj (int) could be propagated to fx.node (ex: sym_size). Is here intentionally blocking the usage? If so, could you suggest an idiomatic way to do it?
